CMSIS/Core/Include/cmsis_compiler.h

```diff
@@ -53,6 +53,21 @@
  */
 #elif defined ( __ICCARM__ )
 
+  /* CMSIS compiler control architecture macros */
+  #if (__CORE__ == __ARM6M__) || (__CORE__ == __ARM6SM__)
```

Comment:
OK let me know once we get IAR statement. I can do the second and not fully certain about the third point.

> @0xc0170 check whether it is possible to exclude this block for compiler versions 8.10 and later. this would highlight the backward compatibility reason of those defines.

We could do it backward-forward compatible. If I move that block as you suggested below cmsis_iar inclusion, then we just check if not yet defined those required macros, and we define them. This should be backward compatible, and forward compatible.

> @0xc0170 check whether it is possible to exclude this block for compiler versions 8.10 and later. this would highlight the backward compatibility reason of those defines.

I can add a comment to this new addition that newer version of IAR (8.x) already define these macros, but older version dont. To explain the reasoning